packages/mdc-select/index.js

```diff
@@ -214,29 +301,148 @@ class MDCSelect extends MDCComponent {
-        addClass: (className) => this.root_.classList.add(className),
-        removeClass: (className) => this.root_.classList.remove(className),
-        hasClass: (className) => this.root_.classList.contains(className),
-        getValue: () => this.nativeControl_.value,
-        isRtl: () => window.getComputedStyle(this.root_).getPropertyValue('direction') === 'rtl',
-        activateBottomLine: () => {
-          if (this.lineRipple_) {
-            this.lineRipple_.activate();
-          }
-        },
-        deactivateBottomLine: () => {
-          if (this.lineRipple_) {
-            this.lineRipple_.deactivate();
-          }
-        },
-      },
-      this.getOutlineAdapterMethods_(),
-      this.getLabelAdapterMethods_())
+      /** @type {!MDCSelectAdapter} */ (Object.assign(
+        this.nativeControl_ ? this.getNativeSelectAdapterMethods_() : this.getEnhancedSelectAdapterMethods_(),
+        this.getCommonAdapterMethods_(),
+        this.getOutlineAdapterMethods_(),
+        this.getLabelAdapterMethods_())
       )
     );
   }
 
+  /**
+   * @return {!{
+   *   getValue: function(): string,
+   *   setValue: function(string): string,
+   *   openMenu: function(): void,
+   *   closeMenu: function(): void,
+   *   isMenuOpen: function(): boolean,
+   *   setSelectedIndex: function(number): void,
+   *   setDisabled: function(boolean): void
+   * }}
+   * @private
+   */
+  getNativeSelectAdapterMethods_() {
+    return {
+      getValue: () => this.nativeControl_.value,
+      setValue: (value) => this.nativeControl_.value = value,
+      openMenu: () => {},
+      closeMenu: () => {},
+      isMenuOpen: () => false,
+      setSelectedIndex: (index) => {
+        this.nativeControl_.selectedIndex = index;
+      },
+      setDisabled: (isDisabled) => this.nativeControl_.disabled = isDisabled,
+    };
+  }
+
+  /**
+   * @return {!{
+   *   getValue: function(): string,
+   *   setValue: function(string): string,
+   *   openMenu: function(): void,
+   *   closeMenu: function(): void,
+   *   isMenuOpen: function(): boolean,
+   *   setSelectedIndex: function(number): void,
+   *   setDisabled: function(boolean): void
+   * }}
+   * @private
+   */
+  getEnhancedSelectAdapterMethods_() {
+    return {
+      getValue: () => {
+        const listItem = this.menuElement_.querySelector(strings.SELECTED_ITEM_SELECTOR);
+        if (listItem) {
+          if (listItem.hasAttribute('value')) {
+            return listItem.getAttribute('value');
+          } else {
+            return listItem.textContent.trim();
+          }
+        }
+        return this.selectedText_.textContent.trim();
+      },
+      setValue: (value) => {
+        const element = /** @type {HTMLElement} */ (this.menuElement_.querySelector(`[value="${value}"]`));
+        if (element) {
+          this.setEnhancedSelectedIndex_(this.menu_.items.indexOf(element));
+        } else {
+          this.selectedText_.textContent = value;
```

Comment:
Falling back to just directly setting `selectedText_.textContent` (and falling back to reading it in `getValue`) also opens the floodgates to this not really behaving like a select anymore (i.e. you can pass in any value, not limited to one in the list).

Moreover, `getValue` falls back to reading the list item's `textContent`, but there's no representation of that case in `setValue`, which IIUC means that list items will never get properly selected if you don't include the `value` attribute, so there's very few cases where that `getValue` branch will be useful.

This all makes me wonder if we should just enforce usage of the attribute.

Meanwhile... is `value` invalid on these elements? Should we be using something like `data-value` instead for enhanced select?

Reply:
`value` is valid on `li` elements. 

I'll update the PR to indicate that using `value` is required.

Reply:
Hm. It's valid, but not the way we're using it.

> This integer attribute indicates the current ordinal value of the list item as defined by the `<ol>` element. The only allowed value for this attribute is a number, even if the list is displayed with Roman numerals or letters. List items that follow this one continue numbering from the value set. The value attribute has no meaning for unordered lists (`<ul>`) or for menus (`<menu>`).

Reply:
So it's valid, but it has no meaning. Why shouldn't we use it?

Reply:
![image](https://user-images.githubusercontent.com/398379/46748539-cb645d80-cc81-11e8-9242-572d3d7a58d3.png)

(If you switch to `ol` for argument's sake, it turns into an error about encountering something other than a `-` or digit.)

I expect someone will log this as an issue, requiring us to change the required markup for the component afterwards, if we don't do it initially.